### PR TITLE
Stepper hang fix

### PR DESCRIPTION
**Pull request by nb-rapidia**
### Description

Cleans up the loop that pops the next block in the planner ISR.

### Benefits

Loop no longer can hang forever, and also it is easier to maintain now.

### Configurations

(configuration file matches 2.0.6.1r)

### CODE REVIEW

File: Marlin/src/module/stepper.cpp [hunk 1]
```diff
@@ -1991,31 +1991,21 @@ uint32_t Stepper::block_phase_isr() {
...
-        }
-        else
+        if (!current_block)
         {
-          break;
+          // No more queued movements, so we cannot proceed.
+          return interval;
         }
-
-        // get a new block
```
Comment (thomaslai) [inline Marlin/src/module/stepper.cpp:2016]: this line?

File: Marlin/src/module/stepper.cpp [hunk 2]
```diff
@@ -1991,31 +1991,21 @@ uint32_t Stepper::block_phase_isr() {
...
       {
-        if (TEST(current_block->flag, BLOCK_BIT_SYNC_POSITION)) {
-          _set_position(current_block->position);
-          discard_current_block();
+        // block was not a motion block,
+        // so we must replace it and try another.
+        discard_current_block();
+        current_block = planner.get_current_block();
 
-          continue;
```
Comment (thomaslai) [inline Marlin/src/module/stepper.cpp:2003]: this continue will be triggered first yes?

File: Marlin/src/module/stepper.cpp [hunk 3]
```diff
@@ -1991,31 +1991,21 @@ uint32_t Stepper::block_phase_isr() {
...
-        }
-        else if (current_block->step_event_count == 0) {
```
Comment (thomaslai) [inline Marlin/src/module/stepper.cpp:2005]: does this block ever get called?

Comment (nb-rapidia) [reply Marlin/src/module/stepper.cpp:2005]: the pause routine sets extrusion-only blocks to have 0 step events.

**Review by thomaslai — APPROVED** [review]

**Comment by nb-rapidia** [comment]
Squashed and merged.

**Pull request closed without merging**